androidtest/src/androidTest/java/com/tozny/e3db/crypto/AndroidConfigHelperTest.java

```diff
@@ -0,0 +1,358 @@
+package com.tozny.e3db.crypto;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.util.Log;
+import com.tozny.e3db.Config;
+import com.tozny.e3db.ConfigStorageHelper;
+import junit.framework.AssertionFailedError;
+import org.junit.Test;
+
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import java.io.File;
+import java.lang.reflect.Method;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+
+public class AndroidConfigHelperTest {
+    private static final String TAG = "AndroidConfigHelperTest";
+
+    private static final String name  = "foo";
+    private static final String loc   = "bar";
+    private static final String alias = "baz";
+
+    private static boolean fileExists(Context context, String privateFile) {
+        File filesDir = context.getFilesDir();
+        return new File(filesDir, privateFile).exists();
+    }
+
+    @Test
+    public void testFileSystemKeystore() {
+        Context context = InstrumentationRegistry.getTargetContext();
+
+        try {
+            if (fileExists(context, name) || fileExists(context, loc)) {
+                File file = new File(context.getFilesDir(), name);
+                file.delete();
+                file = new File(context.getFilesDir(), loc);
+                file.delete();
+            }
+
+            assertFalse(fileExists(context, name));
+            assertFalse(fileExists(context, loc));
+
+            Class<?> clazz = Class.forName("com.tozny.e3db.crypto.FSKSWrapper");
```

Comment:
Probably worth making these package-protected (if possible) so we can call them directly rather than via reflection.